common/src/main/java/com/mx/path/core/common/connect/RetryConfiguration.java

```diff
@@ -0,0 +1,359 @@
+   *
+   * @return non-retying instance
+   */
+  public static RetryConfiguration defaults() {
+    return builder()
+        .stopStrategy(StopStrategy.COUNT)
+        .count(1)
+        .build();
+  }
+
+  /**
+   * Bypass configuration and wrap existing retryer
+   * @param instance
+   */
+  public RetryConfiguration(Retryer instance) {
+    this.instance = instance;
+  }
+
+  // -----------------------------------
+  // Pause settings
+  // -----------------------------------
+
+  /**
+   * Strategy used to calculate the wait between failed attempts
+   *
+   * <p>Default: no pause between failed attempts
+   */
+  @ConfigurationField
+  private PauseStrategy pauseStrategy;
+
+  /**
+   * The pause duration used between failed attempts
+   *
+   * <p>Used by {@link PauseStrategy#FIXED}
+   */
+  @ConfigurationField
+  private Duration pause;
+
+  /**
+   * The first pause duration. Incremented by {@link #increment} between all subsequent failures.
+   *
+   * <p>Used by {@link PauseStrategy#INCREMENTING}
+   */
+  @ConfigurationField
+  private Duration initialPause;
+
+  /**
+   * The amount the pause duration is increased between failures. {@link #initialPause} is used after the first failure.
+   *
+   * <p>Used by {@link PauseStrategy#INCREMENTING}
+   */
+  @ConfigurationField
+  private Duration increment;
+
+  /**
+   * The maximum pause duration
+   *
+   * <p>Used by {@link PauseStrategy#FIBONACCI}
+   */
+  @ConfigurationField
+  private Duration maxPause;
+
+  /**
+   * Duration that is multiplied by the current fibonacci number to get the next pause
+   *
+   * <p>Used by {@link PauseStrategy#FIBONACCI}
+   */
+  @ConfigurationField
+  private Duration multiplier;
+
+  // -----------------------------------
+  // Retry stop settings
+  // -----------------------------------
+  @ConfigurationField(required = true)
+  private StopStrategy stopStrategy;
+
+  /**
+   * The number of failed attempts before stopping
+   *
+   * <p>Used in {@link StopStrategy#COUNT}
+   */
+  @ConfigurationField
+  private Integer count;
+
+  /**
+   * The delay before stopping, starting from first attempt
+   *
+   * <p>Used in {@link StopStrategy#DURATION}
+   */
+  @ConfigurationField
+  private Duration duration;
+
+  // -----------------------------------
+  // Rejection settings
+  // -----------------------------------
+  @ConfigurationField(elementType = ResponseMatcher.class)
+  @Builder.Default
+  private List<ResponseMatcher> retryOn = new ArrayList<>();
+
+  @Getter
+  @Setter
+  private transient Function<Throwable, Throwable> exceptionSupplier;
+
+  private transient Retryer<? extends Response<?, ?>> instance;
+
+  /**
+   * Execute callable block using configured retryer
+   *
+   * @param callable block to call
+   * @return result of first successful attempt
+   * @param <T>
+   * @throws ExecutionException thrown when called throws an exception
+   * @throws RetryException thrown when no attempts succeed
+   */
+  @SneakyThrows
```

Comment:
Do we want to leverage SneakyThrows here?

Reply:
Yes, we need to use it here because we don't know what kind of exception will be thrown by the exception supplier.